src/data/report.ts

```
@@ -153,83 +229,72 @@ export function activate(context: vscode.ExtensionContext, disposables?: Array<v
+
+        const files = [...rootFiles, ...lockFiles]
         showStatus('Creating Socket Report...')
-        const child = child_process.spawn(
-            process.execPath,
-            [
-                entryPoint,
-                'report', 'create', '--json', ...files.map(file => {
-                    const joined = vscode.Uri.joinPath(file.uri, '..')
-                    return joined.fsPath
-                })
-            ],
-            {
-                cwd: workspaceFolderURI.fsPath,
-                env: {
-                    ...process.env,
-                    SOCKET_SECURITY_API_KEY: `${apiKey}`
-                }
-            }
-        )
-        const stdout = text(child.stdout);
-        const stderr = text(child.stderr);
-        try {
-            const [exitCode] = await once(child, 'exit');
-            if (exitCode !== 0) {
-                showErrorStatus((await stderr) || `Failed to run socket reporter child process (exit code ${exitCode})`);
-                return;
+
+        const form = new FormData();
+        for (const file of files) {
+            const filepath = workspace.asRelativePath(file.uri)
+            const fileObj = new File([file.body], filepath)
+            form.set(filepath, fileObj);
+        }
+        const reportBody = new FormDataEncoder(form);
+
+        const req = https.request(`https://api.socket.dev/v0/report/upload`, {
+            method: 'PUT',
+            headers: {
+                ...reportBody.headers,
+                'Authorization': authorizationHeaderValue,
             }
-        } catch (e) {
-            showErrorStatus(e)
-            throw e;
+        });
+        Readable.from(reportBody).pipe(req);
+
+        const [res] = (await once(req, 'response')) as [IncomingMessage]
+        const reportInfo = JSON.parse(await text(res));
+        if (res.statusCode != 200) {
+            showErrorStatus(reportInfo.error.message);
+            return;
```

this should remain in a try/catch and keep the `showErrorStatus` so things like being offline don't silently fail we can at least show the user some info to add an issue to the extension repository.